examples/open_llama/user_script.py

```diff
@@ -4,20 +4,22 @@
 # --------------------------------------------------------------------------
 import torch
 
+from olive.constants import Framework
```

Comment:
I'm okay to keep this for this example for now. But we should be aware of using Olive internal code in user script.

Reply:
IMO, We don't want to use this path.

Reply:
![image](https://github.com/microsoft/Olive/assets/13343117/b57131fe-55f3-48f9-ac20-a58417a5a83f)

For customized evaluation function, we face the same question about using Olive code in user script.

I may think it is fine(not a strong preference) but we should maintain a nice tutorial to tell user how to use.

From user's side, I may think that there is not a big gap better separate user config and some internal Olive flags. 
But from Olive's side, as model type(torch, onnx, snpe, openvino and etc.) increasing, the separate user config may contain:
1. [model_type]_dataloader_func
2. [model_type]_evaluation_func
3. [model_type]_model_loader
In this way, olive have to maintain more similar configs for different model types.